src/ray/design_docs/actor_states.rst

```diff
@@ -0,0 +1,51 @@
+Actor State: Definitions & Transition Diagram
+============================================
+
+An actor can be in one of the following states:
+
+- **DEPENDENCIES_UNREADY**: the actor info is registered in GCS. But its dependencies are not ready.
+
+- **PENDING_CREATION**: the actor local dependencies are ready. This actor is being created.
+
+- **ALIVE**: the actor is created successfully.
+
+- **RESTARTING**: the actor is dead, now being restarted. After reconstruction finishes,
+  the state will become alive again.
+
+- **DEAD**: the actor is already dead and won't be restarted.
+
+::
+
+                                                                         3
+   0                            1                       2          ------------->
+ ---->DEPENDENCIES_UNREADY-------->PENDING_CREATION-------->ALIVE                RESTARTING
+               |                            |                  |   <-------------      |
+               |                            |                  |         4             |
+               |                            |                  |                       |
+             8 |                         7  |                6 |                       | 5
+               |                            |                  |                       |
+               |                            |                  |                       |
+               |                            |                  |                       |
+               |                            v                  |                       |
+                -------------------------->DEAD<---------------------------------------
+
+- **0**: When GCS receives a `RegisterActor` request from core worker, GCS will persist the actor in database with state `DEPENDENCIES_UNREADY`.
+
+- **1**: When core worker has finished resolving the actor dependencies, it will send a `CreateActor` request to GCS and GCS will update actor state to `PENDING_CREATION` in memory.
+
+- **2**: When core worker has finished actor creation task, it will send a `PushTask` reply to GCS and GCS will persist the actor in database with state `ALIVE`.
+
+- **3**: When GCS detected that the worker/node of actor is dead and the actor remaining restarts number is greater than 0, it will persist the actor in database with state `RESTARTING`.
```

Comment:
```suggestion
- **3**: When GCS detectes that the worker/node of an actor is dead and the actor's remaining restarts number is greater than 0, it will persist the actor in database with state `RESTARTING`.
```